src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs

```diff
@@ -129,6 +137,21 @@ public int NumberOfServerInstances
             }
         }
 
+        private void ValidateRemotePipeUser()
+        {
+            if (!IsCurrentUserOnly)
+                return;
+
+            SecurityIdentifier currentUserSid = WindowsIdentity.GetCurrent().Owner;
```

Comment:
Don't forget to dispose the `WindowsIdentity` instance returned by `WindowsIdentity.GetCurrent()`. The GC will eventually reclaim it, but we should be good citizens here. :)